src/types/configmap.ts

```diff
@@ -0,0 +1,13 @@
+import { K8sResourceCommon } from './k8s';
+
+export type ConfigMap = K8sResourceCommon & {
+  kind: string;
+  apiVersion: string;
+  metadata: {
+    name: string;
+    namespace: string;
+  };
```

Comment:
K8sResourceCommon already has these types defined. There is no need to define them again.